serde/src/main/java/com/amazon/ionhiveserde/objectinspectors/IonStructToMapObjectInspector.java

```diff
@@ -80,7 +81,12 @@ public Object getMapValueElement(final Object data, final Object key) {
 
         final Map<String, IonValue> map = new HashMap<>(size, DEFAULT_LOAD_FACTOR);
         for (IonValue v : struct) {
-            map.put(v.getFieldName(), v);
+            if (!IonUtil.isIonNull(v)) {
+                map.put(v.getFieldName(), v);
+            } else {
+                // Hive can't handle Ion null
+                map.put(v.getFieldName(), null);
```

Comment:
With a common `handleNull` function, this can become:
```java
        return StreamSupport.stream(struct.spliterator(), false)
                .collect(HashMap::new, (m,v) -> m.put(v.getFieldName(), handleNull(v)), HashMap::putAll);
```

Beauty is in they eye of the beholder, this may not be preferred. `Collectors.toMap` would look clean but can't be used because it doesn't allow null values. 

If nothing else we can collapse the conditional.
```java
            map.put(v.getFieldName(), handleNull(v));
```

Reply:
Thanks for the example, addressed. I tried to find a workaround to avoid that NPE but couldn't.